app/src/main/java/fr/free/nrw/commons/upload/mediaDetails/UploadMediaDetailFragment.java

```diff
@@ -515,29 +550,47 @@ public void onActivityResult(final int requestCode, final int resultCode,
                 final String longitude = String.valueOf(cameraPosition.target.getLongitude());
                 final double zoom = cameraPosition.zoom;
 
-                editLocation(latitude, longitude,zoom);
+                editLocation(latitude, longitude, zoom);
                 /*
                        If isMissingLocationDialog is true, it means that the user has already tapped the
                        "Next" button, so go directly to the next step.
                  */
-                if(isMissingLocationDialog){
+                if (isMissingLocationDialog) {
                     isMissingLocationDialog = false;
                     onNextButtonClicked();
                 }
             }
         }
+        if (requestCode == REQUEST_CODE_FOR_EDIT_ACTIVITY && resultCode == RESULT_OK) {
+            String result = data.getStringExtra("editedImageFilePath");
+
+            if (!Objects.equals(result, "Error")) {
```

Comment:
Could you please use the Guard Clause pattern here?
https://deviq.com/design-patterns/guard-clause
```
if (Objects.equals(result, "Error")) {
  log error and exit the block
}

try {
  etc
```
or similar